src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -52,40 +56,62 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     } catch (GoogleJsonResponseException e) { // some other problem with requesting comment data
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, COMMENTS_FAILED_ERROR);
     }
-    String comments = String.join(". ", commentsList);
 
     String captions = captionService.getCaptionFromId(videoId);
 
-    if (comments.isEmpty() && captions.isEmpty()) {
-      VideoAnalysis videoAnalysis = new VideoAnalysis.Builder()
-          .setScore(null)
-          .setScoreAvailable(false)
-          .build();
-      String json = new Gson().toJson(videoAnalysis);
-
+    if (commentsList.isEmpty() && captions.isEmpty()) {
+      String json = createResponseJson(null, false);
       response.setContentType("application/json;");
       response.getWriter().println(json);
       return;
     }
+    
+    ExecutorService executor = Executors.newFixedThreadPool(26);
 
-    String text = comments + ". " + captions;
+    List<Future> commentScoreFutures = new ArrayList<>();
+    for (String comment : commentsList) {
+      Future<Float> future = executor.submit(new Callable<Float>() {
+        @Override
+        public Float call() {
+          return calculateSentimentScore(comment);
+        }
+      });
+      commentScoreFutures.add(future);
+    }
 
-    Float score = null;
+    Future<Float> captionFuture = executor.submit(new Callable<Float>() {
+      @Override
+      public Float call() {
+        return calculateSentimentScore(captions);
+      }
+    });
+    
+    Float commentsScore = 0f;
+    Float captionsScore = null;
 
     try {
-      score = calculateSentimentScore(text);
-    } catch (ApiException e) {
+      for (Future<Float> future : commentScoreFutures) {
+        commentsScore += future.get();
+      }
+      commentsScore = commentsScore / commentsList.size();
+      captionsScore = captionFuture.get();
+    } catch (ExecutionException | InterruptedException e) {
       System.err.println(e.getMessage());
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, NLP_API_ERROR);
       return;
     }
 
-    VideoAnalysis videoAnalysis = new VideoAnalysis.Builder()
-        .setScore(score)
-        .setScoreAvailable(true)
-        .build();
-    String json = new Gson().toJson(videoAnalysis);
+    executor.shutdown();
 
+    String json = "";
+    if (!commentsList.isEmpty() && !captions.isEmpty()) {
+      json = createResponseJson((commentsScore + captionsScore) / 2f, true);
```

Comment:
Maybe add some comments for this class on how the scores are calculated, so that they are easier to find?